src/kOS/Communication/VesselConnection.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Communication;
+using UnityEngine;
+using kOS.Safe.Exceptions;
+using kOS.Safe.Serialization;
+using kOS.Suffixed;
+using TimeSpan = kOS.Suffixed.TimeSpan;
+using kOS.Safe.Communication;
+
+namespace kOS.Communication
+{
+    [kOS.Safe.Utilities.KOSNomenclature("Connection", KOSToCSharp = false)]
+    public class VesselConnection : Connection<kOS.SharedObjects>
+    {
+        private Vessel vessel;
+
+        public override bool Connected {
+            get {
+                return shared.ConnectivityMgr.GetDelay(shared.Vessel, vessel) != -1;
+            }
```

Comment:
For the greatest degree of "put all the code in just one place and don't repeat anything" design, perhaps instead of calling ``shared.ConnectivityMgr.GetDelay()``, this should actually call the ``Delay`` Property below in this same class, which in turn calls the ConnectivityMgr.